frontend/src/components/activePage/TableWorkspaceLogic/TableWorkspaceLogic.tsx

```diff
@@ -36,178 +32,101 @@ export interface ITableWorkspaceLogicProps {
-          } = spec?.templateCrownlabsPolitoItTemplateRef?.templateWrapper?.itPolitoCrownlabsV1alpha2Template?.spec!;
-          const {
-            firstName,
-            lastName,
-          } = spec?.tenantCrownlabsPolitoItTenantRef?.tenantWrapper?.itPolitoCrownlabsV1alpha1Tenant?.spec!;
-          return {
-            id: index,
-            name: metadata?.name,
-            gui: environmentList?.[0]?.guiEnabled,
-            persistent: environmentList?.[0]?.persistent,
-            idTemplate: spec?.templateCrownlabsPolitoItTemplateRef?.name!,
-            templatePrettyName: templateName,
-            ip: instance?.status?.ip,
-            status: instance?.status?.phase,
-            url: status?.url,
-            timeStamp: metadata?.creationTimestamp,
-            tenantId: spec?.tenantCrownlabsPolitoItTenantRef?.tenantId,
-            tenantNamespace: `tenant-${spec?.tenantCrownlabsPolitoItTenantRef?.tenantId}`,
-            tenantDisplayName: `${firstName}\n${lastName}`,
-            workspaceId: spec?.templateCrownlabsPolitoItTemplateRef?.namespace,
-          } as Instance;
-        }
-      );
-      const templates = templateList.map(t => {
-        const instancesTmp = instancesMapped?.filter(
-          ({ templatePrettyName: tpn }) => tpn === t
-        );
-        return {
-          id: instancesTmp![0].idTemplate,
-          name: t,
-          gui: instancesTmp![0]?.gui,
-          persistent: instancesTmp![0]?.persistent,
-          resources: { cpu: 0, memory: '', disk: '' },
-          instances: instancesTmp,
-          workspaceId: instancesTmp![0].workspaceId,
-        } as Template;
+          const newItem = { ...prev };
+          setDataInstances(newItem);
+          return prev;
+        },
       });
-      setTemplatesMapped(templates);
     }
-  }, [
-    loadingInstances,
-    dataInstances?.instanceList,
-    userId,
-    errorInstances,
-    workspaces,
-    dataInstances,
-    tenantNamespace,
-  ]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [loadingInstances, subscribeToMoreInstances, tenantNamespace, userId]);
+
+  const instancesMapped = dataInstances?.instanceList?.instances
+    ?.map((i, n) => getManagerInstances(i!, n))
+    .filter(instance => filterId(instance, filter));
+
+  const templatesMapped = getTemplatesMapped(instancesMapped!);
 
   return !loadingInstances && !errorInstances && templatesMapped ? (
     <TableWorkspace
-      workspaces={workspaces.map(ws => ({
-        id: ws.id,
-        title: ws.prettyName,
-        role: ws.role,
-        templates: templatesMapped.filter(
-          ({ workspaceId: id }) => id!.replace(/workspace-/g, '') === ws.id
-        ),
-      }))}
-      filter={''}
+      workspaces={workspaces
+        .map(ws => ({
+          id: ws.id,
+          title: ws.prettyName,
+          role: ws.role,
+          templates: templatesMapped.filter(
+            ({ workspaceId: id }) => id!.replace(/workspace-/g, '') === ws.id
```

Comment:
```suggestion
            ({ workspaceId: id }) => id!.replace(/^workspace-/, '') === ws.id
```
Be careful. This regex could not work properly. If you put `g` it will replace all the occurrences found in the string. However a workspce like `workspace-cool-workspace-yea` would end up broken by this regex.
`^` enforces the match is at the beginning of the string. `g` is not required.
Also, consider making an array of "cleaned up" workspaces names earlier, like right after after `getTemplatesMapped` calling (this is valid for each query which produces workspaces namespaces instead of workspaces names.